File: src/main/java/foodanalysis/Properties.java
package foodanalysis;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

@Data
@Configuration
@ConfigurationProperties(prefix = "main")
public class Properties {

    private TelegramBot telegramBot;
    private SocksProxy socksProxy;

    @Data
    public static class TelegramBot {
        private String username;
        private String token;
    }

    @Data
    public static class SocksProxy {
        private String host;
        private int port;
        private String username;
        private String password;
    }
}

File: src/main/java/foodanalysis/Properties.kt
package foodanalysis

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Configuration

@Configuration
@ConfigurationProperties(prefix = "main")
open class Properties {

    val telegramBot = TelegramBot()

    class TelegramBot {
        lateinit var username: String
        lateinit var token: String
    }

    val socksProxy = SocksProxy()

    class SocksProxy {
        var host: String? = null
        var port = 0
        var username: String? = null
        var password: String? = null
    }
}
